scripts/merge-static-libs.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+#
+# Merge multiple static libraries into one.
+#
+#     scripts/merge-static-libs.sh libtarget.a [libdep1.a ...]
+#
+# Libraries libdep1.a are merged into libtarget.a. That is, all object files
+# from those libraries are copied into the target library.
+#
+# Note that no deduplication is performed so use this script only once.
+
+set -euo pipefail
+
+AR=${AR:-ar}
+
+help() {
+    cat $0 | awk 'NR == 3, /^$/ { print substr($0, 3) }'
+}
+
+while [[ $# -gt 0 ]]
+do
+    case "$1" in
+      -h|--help) help; exit;;
+      --) shift; break;;
+      -*)
+        echo >&2 "Unknown option: $1"
+        echo >&2
+        help
+        exit 1
+        ;;
+      *)
+        break
+        ;;
+    esac
+done
+
+if [[ $# -lt 1 ]]
+then
+    echo >&2 "Invalid command-line: missing target library name"
```

Comment:
> It looks like the script is useless without at least two arguments: a target and minimum one source library.

Not so useless, it is intentionally made a no op if there are no libraries to merge into the target library. This makes its behavior similar to sum function. It is useful because the case when there are no embedded libraries is no longer a special case.

The makefile filters out `*.a` files from dependencies of `libsoter.a` and passes them to this script. If there are no `*.a` files in dependencies (e.g., when system OpenSSL is used) then `merge-static-libs.sh` is invoked with only `build/libsoter.a` and does nothing.

If this case is treated as an error, the Makefile will need to have a conditonal to avoid using `merge-static-libs.sh` which is unnecessary complexity.

Reply:
Thanks for the explanation.